CustomTransition/Application/src/main/java/com/example/android/customtransition/MainActivity.java

```diff
@@ -73,17 +75,16 @@ public boolean onPrepareOptionsMenu(Menu menu) {
 
     @Override
     public boolean onOptionsItemSelected(MenuItem item) {
-        switch(item.getItemId()) {
-            case R.id.menu_toggle_log:
-                mLogShown = !mLogShown;
-                ViewAnimator output = (ViewAnimator) findViewById(R.id.sample_output);
-                if (mLogShown) {
-                    output.setDisplayedChild(1);
-                } else {
-                    output.setDisplayedChild(0);
-                }
-                supportInvalidateOptionsMenu();
-                return true;
+        if (item.getItemId() == R.id.menu_toggle_log) {
+            mLogShown = !mLogShown;
+            ViewAnimator output = findViewById(R.id.sample_output);
+            if (mLogShown) {
+                output.setDisplayedChild(1);
```

Comment:
nitpick. can we inline the condition checking, probably cleaner.

output.setDisplayedChild(if (mLogShown) 1 else 0)

Reply:
Done. This is Java, though.